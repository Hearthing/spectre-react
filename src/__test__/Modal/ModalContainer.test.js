import React from 'react'
import { shallow } from 'enzyme'

import { ModalContainer } from '../..'

describe('ModalContainer', () => {
  it('Should render default .modal-container markup', () => {
    const wrapper = shallow(<ModalContainer />)

    expect(wrapper.type()).toBe('div')
    expect(wrapper.hasClass('modal-container')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <ModalContainer>
        <p>hello world</p>
      </ModalContainer>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<ModalContainer className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('modal-container')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<ModalContainer renderAs='main'>custom</ModalContainer>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('modal-container')).toBe(true)
    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<ModalContainer />)

    expect(wrapper).toMatchSnapshot()
  })
})
