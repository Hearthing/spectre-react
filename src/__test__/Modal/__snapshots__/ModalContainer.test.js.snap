// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContainer Should match the snapshot 1`] = `
<div
  className="modal-container"
/>
`;
